src/tools/ShapeSelect.coffee

```diff
@@ -0,0 +1,85 @@
+{Tool} = require './base'
+{createShape} = require '../core/shapes'
+
+module.exports = class ShapeSelect extends Tool
+  name: 'ShapeSelect'
+  usesSimpleAPI: false
+
+  constructor: (lc) ->
+    # This is a 'shadow' canvas -- we'll reproduce the shapes here, each shape
+    # with a different color that corresponds to their index. That way we'll
+    # be able to find which shape to select on the main canvas by pixel color.
+    @selectCanvas = document.createElement('canvas')
+    @selectCanvas.style['background-color'] = 'transparent'
+    @selectCtx = @selectCanvas.getContext('2d')
+
+  didBecomeActive: (lc) ->
+    shapeSelectUnsubscribeFuncs = []
+    @_shapeSelectUnsubscribe = =>
+      for func in shapeSelectUnsubscribeFuncs
+        func()
+
+    onDown = ({ x, y }) =>
+      @didDrag = false
+
+      shapeIndex = @_getPixel(x, y, lc, @selectCtx)
+      lc.selectShape lc.shapes[shapeIndex]
+
+      if lc.selectedShape?
+        @initialXY = { x, y }
+
+    onDrag = ({ x, y }) =>
+      if lc.selectedShape?
+        @didDrag = true
+
+        xDiff = @initialXY.x - x
+        yDiff = @initialXY.y - y
+        @initialXY = { x, y }
+
+        lc.selectedShape.move { xDiff, yDiff }
+        lc.repaintLayer 'main'
+
+    onUp = ({ x, y }) =>
+      if @didDrag
+        @didDrag = false
+        lc.trigger('shapeMoved', { shape: lc.selectedShape })
+        lc.trigger('drawingChange', {})
+        lc.repaintLayer('main')
+        @_drawSelectCanvas(lc)
+
+    shapeSelectUnsubscribeFuncs.push lc.on 'lc-pointerdown', onDown
+    shapeSelectUnsubscribeFuncs.push lc.on 'lc-pointerdrag', onDrag
+    shapeSelectUnsubscribeFuncs.push lc.on 'lc-pointerup', onUp
+
+    @_drawSelectCanvas(lc)
+
+  willBecomeInactive: (lc) ->
+    @_shapeSelectUnsubscribe()
+    lc.selectShape null
+
+  _drawSelectCanvas: (lc) ->
+    @selectCanvas.width = lc.canvas.width
+    @selectCanvas.height = lc.canvas.height
+    @selectCtx.clearRect(0, 0, @selectCanvas.width, @selectCanvas.height)
+    shapes = lc.shapes.map (shape, index) =>
+      createShape('SelectionBox', {
+        shape: shape,
+        handleSize: 0,
+        backgroundColor: "##{@_intToHex(index)}"
+      })
```

Comment:
So in theory, this could also return a copy of the shape if you wanted to select by pixels rather than bounding box? (Don't need to make it configurable now, but it's good to know.)